File: testing/corpora/golden_corpus/08_wrapper_apis/wrapper_api.py
"""
Category: Wrapper APIs (Phase 22.3 Golden Corpus)
Contains custom enterprise wrapper libraries and abstraction facades
delegating to underlying cryptographic libraries.
"""

import hashlib
from cryptography.hazmat.primitives.ciphers.aead import AESGCM


class EnterpriseCryptoFacade:
    """Internal enterprise wrapper hiding direct crypto calls."""

    def __init__(self, key: bytes):
        # Wraps AESGCM 256-bit
        self._engine = AESGCM(key)
        self._default_hash = "sha256"

    def secure_storage_encrypt(self, record_payload: bytes, nonce: bytes) -> bytes:
        """Custom wrapper method delegating to AES-GCM."""
        return self._engine.encrypt(nonce, record_payload, None)

    def compute_integrity_checksum(self, buffer_data: bytes) -> str:
        """Custom wrapper method delegating to hashlib.sha256."""
        return hashlib.sha256(buffer_data).hexdigest()

    def generate_token_digest(self, token_str: str) -> str:
        """Custom wrapper method delegating to hashlib.sha384."""
        return hashlib.sha384(token_str.encode("utf-8")).hexdigest()

File: testing/corpora/golden_corpus/08_wrapper_apis/wrapper_api.ts
/**
 * Category: Wrapper APIs (Phase 22.3 Golden Corpus)
 * TypeScript custom corporate security facade
 */

import * as crypto from 'crypto';

export interface CryptoWrapperService {
  encryptField(data: string, secretKey: Buffer): string;
  hashIdentifier(id: string): string;
}

export class CorporateSecurityService implements CryptoWrapperService {
  public encryptField(data: string, secretKey: Buffer): string {
    const iv = crypto.randomBytes(12);
    const cipher = crypto.createCipheriv('aes-256-gcm', secretKey, iv);
    let enc = cipher.update(data, 'utf8', 'hex');
    enc += cipher.final('hex');
    return `${iv.toString('hex')}:${enc}`;
  }

  public hashIdentifier(id: string): string {
    return crypto.createHash('sha256').update(id).digest('hex');
  }
}
